### DISCUSSION TITLE
Roku Remote Entity Id in YAML config

### DISCUSSION THREAD

**Discussion by DennisOtt-git-hub** [PRProd/HA-Firemote#574] [Q&A] [answered]
I cannot find a way to set the Associated Remote to the Roku remote YAML configuration.  It appears to be an available setting in the HTML dashboard input, but I use YAML configurations of my dashboards and cannot find the entry to set this so all the non-app launcher buttons return an 'missing entity id' error message.  The only option seems to be defining button overrides.  But this seems like it should be a lot of unnecessary work and avoidable if the remote entity id would be available.  

Any help would be appreciated.

**Comment by PRProd** [comment, marked as answer]
Greetings @DennisOtt-git-hub!  It's nice to have you here!

The instructions you are looking for are in the [README.md](https://github.com/PRProd/HA-Firemote/blob/main/README.md) file in a section called [YAML card setup options](https://github.com/PRProd/HA-Firemote?tab=readme-ov-file#yaml-card-setup-options)

![image](https://github.com/user-attachments/assets/04f762f0-ba58-4173-8ecc-9c22132e63bd)
![image](https://github.com/user-attachments/assets/1985c14e-2e0e-46da-af6c-1092d8a92d47)

Depending on which Roku device you own, you can choose the proper option from this list and add that entry to your YAML config like this:
```yaml
device_family: roku
device_type: roku-ultra-lt-2023
```
<br>
<br>
<br>

The ``missing entity id`` message could be a result of this missing entry in your YAML config
![image](https://github.com/user-attachments/assets/0721e743-dcfa-4e7e-a8f0-5d49a6900430)

In my environment, that YAML entry looks like this:
```yaml
entity: media_player.roku_ultra
```
The same is true for the roku_remote_entity:
```yaml
roku_remote_entity: remote.roku_ultra
```

Yours will look different - where mine says 'media_player.roku_ultra' and 'remote.roku_ultra', you will want to replace that with the media player entity id of your Roku device.

I hope this helps!  Please let me know!

-Doug

**Reply by PRProd** [reply]
Sorry about that missing roku_remote_entity information in the readme.  I've just updated the documentation to include that one.
